dev/service-registry/client-core/src/main/scala/com/lightbend/lagom/internal/registry/DevModeSimpleServiceDiscovery.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.internal.registry
+
+import java.net.URI
+
+import akka.actor.ActorSystem
+import akka.discovery.SimpleServiceDiscovery._
+import akka.discovery.{ Lookup, ServiceDiscovery, SimpleServiceDiscovery }
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ ExecutionContext, Future, Promise }
+
+private[lagom] class DevModeSimpleServiceDiscovery(system: ActorSystem) extends SimpleServiceDiscovery {
+  private val clientPromise = Promise[ServiceRegistryClient]
+
+  implicit private val ec: ExecutionContext = system.dispatcher
+
+  def setServiceRegistryClient(client: ServiceRegistryClient): Unit = clientPromise.success(client)
```

Comment:
I missed out on why it needs to be set after the construction instead of injected in the constructor, both places it is created you already have the client when the `DevModeSimpleServiceDiscovery` is created?

Reply:
Aha! I missed that and thought it was just a case class at the lookup site (which I thought was construction site). Makes sense then.